### PR TITLE
Move `traitsui.qt4` to `traitsui.qt`

### PR DESCRIPTION
**Pull request by corranwebster**
Also updates imports etc.  This means that we need Pyface 8.0 or greater.

**Checklist**
- [ ] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

**Comment by corranwebster** [comment]
This really needs Pyface 8.0 to be released to make it work; can patch with github installs for now, but will need to reset before release.

File: .github/workflows/test-with-edm.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 name: Test with EDM
 
-on: [pull_request, workflow_dispatch]
```
Comment (mdickinson) [inline .github/workflows/test-with-edm.yml:11]: Was this change intentional?

Comment (corranwebster) [reply .github/workflows/test-with-edm.yml:11]: Yes.  No point in running these until EDM has Pyface 8.

**Review by mdickinson — APPROVED** [review]
Looks fine to me on a quick skim.

Was the removal of the pull_request trigger for the EDM workflow intentional? There's no mention of it in the PR description, and it seems unrelated to the stated PR purpose.

**Comment by mdickinson** [comment]
> Was the removal of the pull_request trigger for the EDM workflow intentional?

Already answered in a comment above.

**Comment by corranwebster** [comment]
> Was the removal of the pull_request trigger for the EDM workflow intentional? There's no mention of it in the PR description, and it seems unrelated to the stated PR purpose.

I should change it to mention the immediate cause which is that we need pyface >= 8; and I should probably update the pyproject.toml to reflect that.
